### PR TITLE
Add pg_hba rule to allow zabbix server #411

### PR DESCRIPTION
**Pull request by RaphaelNeumann**
Add a pg_hba rule to aways add this rule for allow zabbix server connect to database. #411

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @RaphaelNeumann, thanks for the patch. Can you take a look at the failing spec tests?

**Comment by RaphaelNeumann** [comment]
@bastelfreak of course... I will send fix soon

**Comment by RaphaelNeumann** [comment]
@bastelfreak It's there :D

File: manifests/database.pp [hunk 1]
```diff
@@ -157,7 +155,13 @@
             address     => "${zabbix_server_ip}/32",
             auth_method => 'md5',
           }
+        }
 
+        # When every component has its own server, we have to allow those servers to
+        # access the database from the network. Postgresl allows this via the
```
Comment (bastelfreak) [inline manifests/database.pp]: typo: Postgresl -> Postgresql

Comment (RaphaelNeumann) [inline manifests/database.pp]: thanks man

File: spec/classes/database_spec.rb [hunk 2]
```diff
@@ -65,6 +66,32 @@
         it { is_expected.to contain_class('zabbix::params') }
       end
 
+      describe 'database_type is postgresql, zabbix_type is server and zabbbix_server anda zabbix_web in the some server but zabbix_database is on other server' do
```
Comment (bastelfreak) [inline spec/classes/database_spec.rb]: typo: anda -> "and a"

File: manifests/database.pp [hunk 3]
```diff
@@ -144,11 +145,8 @@
           require  => Class['postgresql::server'],
         }
 
-        # When every component has its own server, we have to allow those servers to
-        # access the database from the network. Postgresl allows this via the
-        # pg_hba.conf file. As this file only accepts ip addresses, the ip address
-        # of server and web has to be supplied as an parameter.
-        if $zabbix_web_ip != $zabbix_server_ip {
+        # When database not in some server with zabbix server include pg_hba_rule to server
+        if ($database_host_ip != $zabbix_server_ip) or ($zabbix_web_ip != $zabbix_server_ip){
```
Comment (bastelfreak) [inline manifests/database.pp]: is this logic correct? Shouldn't it be: ```if ($database_host_ip != $zabbix_server_ip){}``` ?

Comment (RaphaelNeumann) [inline manifests/database.pp]: Yes, I agree with this logic, but on current and old versions of this module has this rule, if I change this logic, current implements will stop working.

For warrant  the retro-compatibility I just added an "or" condition for this existent rule.

**Comment by bastelfreak** [comment]
Thanks!

**Pull request merged**